packages/skygear-sso/lib/oauth.js

```diff
@@ -0,0 +1,231 @@
+/* global window:false document:false */
+import cookies from 'js-cookie';
+import { atob } from 'Base64';
+import { NewWindowObserver, PostAuthResultObserver } from './observer';
+import { errorResponseFromMessage } from './util';
+/**
+ * Calling plugin to retrieve auth url and open in popup
+ *
+ * @injectTo {AuthContainer} as loginOAuthProviderWithPopup
+ * @param  {String} provider - name of provider, e.g. google, facebook
+ * @param  {Object} options - options for generating auth_url
+ * @return {Promise} promise
+ *
+ * @example
+ * skygear.auth.loginOAuthProviderWithPopup('google').then(...);
+ */
+export function loginOAuthProviderWithPopup(provider, options) {
+  var newWindow = window.open('', '_blank', 'height=700,width=500');
+  this._oauthWindowObserver = this._oauthWindowObserver ||
+    new NewWindowObserver();
+  this._oauthResultObserver = this._oauthResultObserver ||
```

Comment:
I think it is impossible for `_oauthResultObserver` having a non-null value here?